salt/template.py

```diff
@@ -88,8 +88,9 @@ def compile_template(template,
         if not isinstance(input_data, (dict, OrderedDict)):
             try:
                 input_data.seek(0)
-            except Exception as exp:
-                log.error('error while compiling template \'{0}\': {1}'.format(template, exp))
+            except AttributeError:
```

Comment:
The only possible other exception that could be thrown here is an IOError, in case the object is seekable, but it fails for some reason. But since this is an in-memory-string handled as a file object, that is (imo) very unlikely to happen. This lead me to conclude that a generic exception is out of place here.
I do not quite understand the comment at https://github.com/saltstack/salt/commit/99024ba1c86dba21550c28f4998aaa0cd0f0169a which could shed some light on why a catch-all is here in the first place (even though it was silenced).